apps/flame-defi/app/pool/utils/sdk-helpers.ts

```diff
@@ -0,0 +1,19 @@
+import { Token } from "@uniswap/sdk-core";
+import type { Address } from "viem";
+
+import type { AstriaChain, EvmCurrency } from "@repo/flame-types";
+
+/**
+ * Converts an internal token instance to a Uniswap SDK token instance.
+ */
+export const getTokenFromInternalToken = (
```

Comment:
Could this be a method on the type, like `EvmCurrency.asUniswapToken(chain)`?

Reply:
What if we actually added an `EvmCurrency.wrapped` property that is also of type `EvmCurrency` (would require refactoring the way the configs are defined is all), along with a new method `asToken`.

You wouldn't even need this method anymore, you'd just called `currency.asToken()` and in `asToken` would return
```typescript
this.isNative ? this.wrapped.asToken() : this.asToken()
```

Uniswap has something similar with `NativeCurrency.wrapped`, and `Token`